libs/schemas/src/entities/user.schemas.ts

```diff
@@ -74,6 +73,7 @@ export const Address = z.object({
   hex: z.string().max(64).optional(),
   created_at: z.any(),
   updated_at: z.any(),
+  User: User.optional(),
```

Comment:
When we had similar props in the `XXXAttributes` types, it was messy and confusing dealing with entity props vs join props.

I wonder if we should somehow separate them?

Sure, we capitalize the join props to somewhat distinguish them from entity props, but would it be cleaner if we maybe did `.extend({ ... })` to semantically convey that they're not part of the base entity?

Reply:
I agree this is confusing but the client is expecting this shape and I added it to the schema to temporarily improve client type safety... in the tRPC refactoring bucket we have to define a cleaner convention in the response schemas.